### PR TITLE
🐛 Add validity checks to egoJwt util funcs, refresh token early

### PR DESCRIPTION
**Pull request by anncatton**
**Description of changes**

Refresh jwt 30sec before expiry
Add validity check to all eogJwt util functions
Remove polling in user query (was for dev testing)

**Type of Change**

- [x] Bug
- [ ] Styling
- [ ] New Feature

**Checklist before requesting review:**

- [ ] Matches design:

  - component sizes, spacing, and styles
  - font size, weight, colour
  - spelling has been double checked

- [ ] New uikit components have Storybook stories
- [ ] Feature is minimally responsive.
- [ ] Manual testing of UI feature.
- [ ] Selenium test is completed and passing.

### CODE REVIEW

File: global/utils/egoJwt/index.tsx [hunk 1]
```diff
@@ -43,60 +42,61 @@ export const decodeToken = memoize(
...
-  !!egoJwt && TokenUtils.canReadSomeProgramData(egoJwt);
+  isValidJwt(egoJwt) && TokenUtils.canReadSomeProgramData(egoJwt);
 
 export const canWriteSomeProgramData: (egoJwt: string) => boolean = egoJwt =>
-  !!egoJwt && TokenUtils.canWriteSomeProgramData(egoJwt);
+  isValidJwt(egoJwt) && TokenUtils.canWriteSomeProgramData(egoJwt);
 
 export const getReadableProgramDataScopes: (egoJwt: string) => PermissionScopeObj[] = egoJwt =>
-  !egoJwt ? [] : TokenUtils.getReadableProgramDataScopes(egoJwt);
+  isValidJwt(egoJwt) ? [] : TokenUtils.getReadableProgramDataScopes(egoJwt);
```
Comment (hlminh2000) [inline global/utils/egoJwt/index.tsx:91]: careful with `!` vs `!!` :P `isValidJwt(egoJwt) ? [] : ...` doesn't make sense for `getReadableProgramDataScopes`

File: global/hooks/useAuthContext.tsx [hunk 2]
```diff
@@ -107,6 +109,15 @@ export function AuthProvider({
       });
   };
 
+  React.useEffect(() => {
+    if (token && isValidJwt(token)) {
+      const expiry = decodeToken(token).exp;
```
Comment (hlminh2000) [inline global/hooks/useAuthContext.tsx:114]: just wanna confirm, `exp` is the expiry timestamp, not time to expiry from issuance right? :P I'm just paranoid...

File: global/utils/egoJwt/index.tsx (hunk 1 above)
Comment (anncatton) [reply global/utils/egoJwt/index.tsx:91]: Whoops - I will correct those 4 I changed

File: global/hooks/useAuthContext.tsx (hunk 2 above)
Comment (anncatton) [reply global/hooks/useAuthContext.tsx:114]: Good call! Double checking

Comment (anncatton) [reply global/hooks/useAuthContext.tsx:114]: Yes it is the `"exp"` value in the jwt

**Pull request merged**